apps/myaccount/src/components/shared/system-notification-alert.tsx

```diff
@@ -0,0 +1,207 @@
+import { SCIMConfigs } from "../../extensions/configs/scim";
+import { AlertLevels, AuthStateInterface } from "../../models";
+import { PreferenceConnectorResponse, PreferenceProperty, PreferenceRequest } from "../../models/preference";
+import { AppState } from "../../store";
+import { addAlert } from "../../store/actions";
+import "./system-notification-alert.scss";
+
+/**
+ * SystemNotificationAlert component props.
+ */
+export type SystemNotificationAlertProps = IdentifiableComponentInterface;
+
+/**
+ * System Notification Alert component.
+ *
+ * @param props - Props injected into the component.
+ * @returns ReactElement
+ */
+export const SystemNotificationAlert: FunctionComponent<SystemNotificationAlertProps> = (
+    props: SystemNotificationAlertProps
+): ReactElement | null => {
+    const { [ "data-componentid" ]: componentId } = props;
+
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const profileDetails: AuthStateInterface = useSelector((state: AppState) => state.authenticationInformation);
+
+    const [ isSRSendOtpInEmailEnabled, setIsSRSendOtpInEmailEnabled ] = useState<boolean>(false);
+    const [ isAccountStatePendingSR, setIsAccountStatePendingSR ] = useState<boolean>(false);
+    const [ isPreferredChannelEmail, setIsPreferredChannelEmail ] = useState<boolean>(false);
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+
+    useEffect(() => {
+        getSelfSignUpPreferences();
+    }, []);
+
+    useEffect(() => {
+        const accountState: string = profileDetails
+            ?.profileInfo
+            ?.[SCIMConfigs?.scim?.systemSchema]
+            ?.[ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get(MyAccountProfileConstants.ACCOUNT_STATE)];
+
+        const preferredChannel: string = profileDetails
+            ?.profileInfo
+            ?.[SCIMConfigs?.scim?.systemSchema]
+            ?.[ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get(MyAccountProfileConstants.PREFERRED_CHANNEL)];
+
+        setIsAccountStatePendingSR(accountState === MyAccountProfileConstants.ACCOUNT_STATE_PENDING_SR);
+        setIsPreferredChannelEmail(preferredChannel === MyAccountProfileConstants.PREFERRED_CHANNEL_EMAIL);
+    }, [ profileDetails?.profileInfo ]);
+
+    /**
+     * Gets the self sign up account confirmation preferences.
+     */
+    const getSelfSignUpPreferences = (): void => {
+        const selfSignUpConnector: PreferenceRequest[] = [ {
+            "connector-name": ProfileConstants.SELF_SIGN_UP_CONNECTOR,
+            properties: [ ProfileConstants.SELF_SIGN_UP_ENABLE_SEND_OTP_IN_EMAIL ]
+        } ];
+
+        getPreference(selfSignUpConnector)
+            .then((response: PreferenceConnectorResponse[]) => {
+                if (response) {
+                    const selfSignUpOptions: PreferenceConnectorResponse[] = response;
+                    const responseProperties: PreferenceProperty[] = selfSignUpOptions[0].properties;
```

Comment:
is taking the first element of `selfSignUpOptions` reliable?

Reply:
Resolved via https://github.com/wso2/identity-apps/pull/8071/commits/ad689a3e1bfb488e07f08dbac7018befc1241bcc